packages/fuels-core/src/abi_encoder.rs

```diff
@@ -139,6 +132,38 @@ impl ABIEncoder {
             Self::encode_u64(len),
         ])
     }
+
+    fn encode_raw_slice(data: &[u64]) -> Result<Vec<Data>> {
+        let encoded_data = data
+            .iter()
+            .map(|&word| Self::encode_u64(word))
+            .collect::<Vec<_>>();
+
+        let num_bytes = data.len() * WORD_SIZE;
+
+        let len = Self::encode_u64(num_bytes as u64);
+        Ok(vec![Data::Dynamic(encoded_data), len])
+    }
+
+    fn encode_bytes(mut data: Vec<u8>) -> Result<Vec<Data>> {
+        let len = data.len();
+
+        Self::zeropad_to_word_alignment(&mut data);
+
+        let cap = data.len() as u64;
+        let encoded_data = vec![Data::Inline(data)];
+
+        Ok(vec![
+            Data::Dynamic(encoded_data),
+            Self::encode_u64(cap),
+            Self::encode_u64(len as u64),
+        ])
+    }
+
+    fn zeropad_to_word_alignment(data: &mut Vec<u8>) {
+        let padded_length = padded_len_usize(data.len());
+        data.resize(padded_length, 0);
```

Comment:
I don't think introducing a function just for this is justified here. The `ABIEncoder` struct only contains `encode_xxx` methods, I don't see why we would add a different method just for this. If the meaning is not clear enough, then a comment can be enough.

Reply:
```suggestion
    fn encode_bytes(mut data: Vec<u8>) -> Result<Vec<Data>> {
        let len = data.len();
        
       // zeropad to word alignement
        let padded_length = padded_len_usize(data.len());
        data.resize(padded_length, 0);
        
        / --- snip ---/
```

is what I had in mind. I just don't like that we're introducing a one line function that isn't used anywhere else. I believe that for functions that short, yes at least two call sites should be the minimum. To me this is factorization to the extreme and I don't understand it.

Reply:
> To me this is factorization to the extreme
 
 To me, it is about hiding details, however small, and talking on a higher abstraction level. The mention of `padded_len_usize`, `resize` are all noise to what I want to communicate to code readers, and that is: we'll zeropad the bytes until word alignment is reached -- go see how if you're that interested.

> we're breaking the nice "all functions inside ABIEncoder are decoding functions"

It's nice to look at, sure. I wouldn't take it as a goal. We're bound to have a helper sooner or later mixed in.

But I've extracted the zeropad fn as a private function after the encoder. I dislike private associated functions anyways. Now we again have all methods start with 'encode_*'.